backend-rust/src/indexer.rs

```diff
@@ -0,0 +1,744 @@
+//! TODO:
```

Comment:
This file also defines many separate types. It'd be easier to get an overview if it was split up a bit more.

Reply:
I think scattering types across files is harder to get an overview of in general, especially if they are tied into some context across files.
But if there are some clear separation of them, it could make sense, but as mentioned above, it should probably be done in the end when changes are less likely

Reply:
If they are tied together, they can still be related to each other via submodules. It doesn't have to be separate modules as siblings, it could be a parent-child relationship.

Reply:
If it makes sense, sure.